hs_core/urls.py

```diff
@@ -92,8 +93,16 @@
     url(r'^resource/(?P<pk>[0-9a-f-]+)/functions/move-or-rename/$',
         views.resource_folder_hierarchy.data_store_file_or_folder_move_or_rename_public),
 
+    # public move
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/functions/move-to-folder/$',
+        views.resource_folder_hierarchy.data_store_move_to_folder_public),
+
+    # public rename
+    url(r'^resource/(?P<pk>[0-9a-f-]+)/functions/rename-file-or-folder/$',
```

Comment:
I am following the UI. And I need to follow it more closely. 

The `rename` function just takes one thing and renames it to another thing **in the UI**. There is no confusion whether or not it is a file or folder.

The `move` function is -- by contrast -- called in the UI with multiple sources and one destination (through 'paste'). This needs to be closer to the UI.